src/@next/pages/CheckoutPage/CheckoutPage.tsx

```diff
@@ -1,398 +1,55 @@
-      onError={handlePaymentGatewayError}
-      errors={paymentGatewayErrors}
-    />
-  );
+  if (!cartLoaded || !checkoutLoaded) {
+    return <Loader />;
+  }
 
-  const handlePaymentConfirm = async () => {
-    /**
-     * Prevent proceeding in confirmation flow in case of gateways that don't support it to prevent unknown bugs.
-     */
-    if (payment?.gateway !== "mirumee.payments.adyen") {
-      const paymentStepLink = steps.find(
-        step => step.step === CheckoutStep.Payment
-      )?.link;
-      if (paymentStepLink) {
-        push(paymentStepLink);
-      }
-    }
+  if (!items || !items.length) {
+    replace("/cart");
+    return null;
+  }
 
-    setSubmitInProgress(true);
-    setPaymentConfirmation(true);
-    /**
-     * Saleor API creates an order for not fully authorised payments, thus we accept all non negative payment result codes,
-     * assuming the payment is completed, what means we can proceed further.
-     * https://docs.adyen.com/checkout/drop-in-web?tab=http_get_1#step-6-present-payment-result
-     */
-    if (
-      adyenNotNegativeConfirmationStatusCodes.includes(
-        query.resultCode as string
-      )
-    ) {
-      const { data, dataError } = await completeCheckout();
-      const errors = dataError?.error;
-      setSubmitInProgress(false);
-      if (errors) {
-        setPaymentGatewayErrors(errors);
-        const paymentStepLink = steps.find(
-          step => step.step === CheckoutStep.Payment
-        )?.link;
-        if (paymentStepLink) {
-          push(paymentStepLink);
-        }
-      } else {
-        setPaymentGatewayErrors([]);
-        handleStepSubmitSuccess(CheckoutStep.Review, {
-          id: data?.order?.id,
-          orderNumber: data?.order?.number,
-          token: data?.order?.token,
-        });
-      }
-    } else {
-      setPaymentGatewayErrors([
-        {
-          message: translateAdyenConfirmationError(
-            query.resultCode as string,
-            intl
-          ),
+  const createCheckout = async () => {
+    try {
+      const email = user?.email || "customer@example.com";
+      const response = await apolloClient.mutate({
+        mutation: createCheckoutQuery,
+        variables: {
+          lines: (items || []).map(item => ({
+            quantity: item.quantity,
+            variantId: item.variant.id,
+          })),
+          email,
         },
-      ]);
-      const paymentStepLink = steps.find(
-        step => step.step === CheckoutStep.Payment
-      )?.link;
-      if (paymentStepLink) {
-        push(paymentStepLink);
-        setSubmitInProgress(false);
-        setPaymentConfirmation(false);
-      }
+      });
+      const newCheckoutToken = response?.data?.checkoutCreate?.checkout?.token;
+      replace(`${paths.checkoutCreateJobAd}/?id=${newCheckoutToken}`);
+    } catch (error) {
+      console.log("Error on creating checkout", error); // eslint-disable-line no-console
```

Comment:
Must the user not be informed as well that something went wrong? Maybe new ticket?

Reply:
> not sure how can we display this kind of errors for now